File: VB/Q556797/Models/MyModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace MVCxGridViewDataBinding.Models
	Public Class MyModel
		Public Sub New()
			Me.New(0)

		End Sub
		Public Sub New(ByVal index As Integer)
			ModelID = index
			ModelName = "Name" & index
			If index Mod 20 = 0 Then

				ModelState = Nothing
				Return
			End If
			If index Mod 10 = 0 Then
				ModelState = ModelStateEnum.None
				Return
			End If
			If index Mod 2 = 0 Then
				ModelState = ModelStateEnum.NonImportant
			Else
				ModelState = ModelStateEnum.Important
			End If

		End Sub
		' Fields...
		Private _ModelID As Integer
		Private _ModelName As String
		Private _ModelState? As ModelStateEnum


		Public Property ModelID() As Integer
			Get
				Return _ModelID
			End Get
			Set(ByVal value As Integer)
				_ModelID = value
			End Set
		End Property
		Public Property ModelName() As String
			Get
				Return _ModelName
			End Get
			Set(ByVal value As String)
				_ModelName = value
			End Set
		End Property
		Public Property ModelState() As ModelStateEnum?
			Get
				Return _ModelState
			End Get
			Set(ByVal value? As ModelStateEnum)
				_ModelState = value
			End Set
		End Property
		Public Shared Function GetModelList() As List(Of MyModel)
			Dim l As New List(Of MyModel)()
			For i As Integer = 0 To 9999
				l.Add(New MyModel(i))
			Next i
			Return l
		End Function
		Public Shared Sub UpdateModel(ByVal l As List(Of MyModel), ByVal m As MyModel)

			Dim editedModel As MyModel = l.Where(Function(x) x.ModelID = m.ModelID).First()
			editedModel.ModelName = m.ModelName
			editedModel.ModelState = m.ModelState
		End Sub
		Public Shared Sub DeleteModel(ByVal l As List(Of MyModel), ByVal mID As Integer)
			l.Remove(l.Where(Function(x) x.ModelID = mID).First())
		End Sub
		Public Shared Sub InsertModel(ByVal l As List(Of MyModel), ByVal m As MyModel)
			m.ModelID = l.Count + 1
			l.Add(m)
		End Sub
	Public Enum ModelStateEnum
			None = 2
			Important = 1
			NonImportant = 0
	End Enum
	End Class
End Namespace

File: CS/Q556797/Models/MyModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MVCxGridViewDataBinding.Models
{
    public class MyModel
    {
        public MyModel()
            : this(0)
        {

        }
        public MyModel(int index)
        {
            ModelID = index;
            ModelName = "Name" + index;
            if (index % 20 == 0)
            {

                ModelState = null;
                return;
            }
            if (index % 10 == 0)
            {
                ModelState = ModelStateEnum.None;
                return;
            }
            if (index % 2 == 0)
                ModelState = ModelStateEnum.NonImportant;
            else
                ModelState = ModelStateEnum.Important;
           
        }
        // Fields...
        private int _ModelID;
        private string _ModelName;
        private ModelStateEnum? _ModelState;
      

        public int ModelID
        {
            get { return _ModelID; }
            set { _ModelID = value; }
        }
        public string ModelName
        {
            get { return _ModelName; }
            set { _ModelName = value; }
        }
        public ModelStateEnum? ModelState
        {
            get { return _ModelState; }
            set { _ModelState = value; }
        }      
        public static List<MyModel> GetModelList()
        {
            List<MyModel> l = new List<MyModel>();
            for (int i = 0; i < 10000; i++)
            {
                l.Add(new MyModel(i));
            }
            return l;
        }
        public static void UpdateModel(List<MyModel> l, MyModel m)
        {

            MyModel editedModel = l.Where(x => x.ModelID == m.ModelID).First();
            editedModel.ModelName = m.ModelName;
            editedModel.ModelState = m.ModelState;
        }
        public static void DeleteModel(List<MyModel> l, int mID)
        {
            l.Remove(l.Where(x => x.ModelID == mID).First());
        }
        public static void InsertModel(List<MyModel> l, MyModel m)
        {
            m.ModelID = l.Count + 1;
            l.Add(m);
        }
    public  enum ModelStateEnum
        {
            None = 2,
            Important = 1,
            NonImportant = 0
        };
    }
}